qutebrowser/browser/greasemonkey.py

```diff
@@ -0,0 +1,289 @@
+        self._code = code
+        self._includes = []
+        self._excludes = []
+        self._description = None
+        self._name = None
+        self._run_at = None
+        for name, value in properties:
+            if name == 'name':
+                self._name = value
+            if name == 'description':
+                self._description = value
+            if name in ['include', 'match']:
+                self._includes.append(value)
+            if name in ['exclude', 'exclude_match']:
+                self._excludes.append(value)
+            if name == 'run-at':
+                self._run_at = value
+
+    HEADER_REGEX = r'\/\/ ==UserScript==.|\n+\/\/ ==\/UserScript==\n'
+    PROPS_REGEX = r'\/\/ @(?P<prop>[^\s]+)\s+(?P<val>.+)'
+
+    @classmethod
+    def parse(cls, source):
+        props, _code = re.split(cls.HEADER_REGEX, source)
+        return cls(re.findall(cls.PROPS_REGEX, props), source)
+
+    def includes(self):
+        return self._includes
+
+    def excludes(self):
+        return self._excludes
+
+    def name(self):
+        return self._name
+
+    def run_at(self):
+        return self._run_at
+
+    def code(self):
+        gm_bootstrap = self.GM_BOOTSTRAP_TEMPLATE.format(
+            scriptName=self._name,
+            scriptInfo=self.meta_json() or 'null',
+            scriptMeta=self.meta_raw() or 'null')
+        return '\n'.join(
+            ["(function(){", gm_bootstrap, self._code, "}).call();"])
+
+    def meta_json(self):
+        return json.dumps({
+            'name': self._name,
+            'description': self._description,
+            'matches': self._includes,
+            'includes': self._includes,
+            'excludes': self._excludes,
+            'run-at': self._run_at,
+        })
+
+    def meta_raw(self):
+        pass
+
+
+class GreasemonkeyManager(QObject):
+
+    """Manager of userscripts and a greasemonkey compatible environment.
+
+    Signals:
+        scripts_reloaded: Emitted when scripts are reloaded from disk.
+            Any any cached or already-injected scripts should be
+            considered obselete.
+    """
+
+    scripts_reloaded = pyqtSignal()
+
+    def __init__(self):
+        super().__init__()
+        self.load_scripts()
+
+    @cmdutils.register(name='greasemonkey-reload',
+                       instance='greasemonkey')
+    def load_scripts(self):
+        """Re-Read greasemonkey scripts from disk."""
+        self._run_start = []
+        self._run_end = []
+        self._run_idle = []
+
+        scripts_dir = QDir(_scripts_dir())
+        log.greasemonkey.debug("Reading scripts from: %s",
+                               scripts_dir.absolutePath())
+        for script_filename in scripts_dir.entryList(['*.js'], QDir.Files):
+            script_path = scripts_dir.absoluteFilePath(script_filename)
+            log.greasemonkey.debug("Trying to load script: %s", script_path)
+            with open(script_path, encoding='utf-8') as script_file:
```

Comment:
I do think `encoding='utf-8'` is the right choice here - the default on Windows would be [latin1](https://en.wikipedia.org/wiki/ISO/IEC_8859-1) (at least in my part of the world) which doesn't cover many languages. I also downloaded some userscripts, and they were UTF-8 encoded.